tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -59,17 +59,28 @@ def test_nodereplacement_proactive(self, pvc_factory, pod_factory, dc_pod_factor
                 cephfs_dc_pod = dc_pod_factory(interface=constants.CEPHFILESYSTEM, node_name=worker_node, size=20)
                 pod.run_io_in_bg(cephfs_dc_pod, expect_to_fail=False, fedora_dc=True)
 
+        # error message for invalid deployment configuration
+        msg_invalid = (
+            "ocs-ci config 'deployment_type' value "
+            f"'{config.ENV_DATA['deployment_type']}' is not valid, "
+            f"results of this test run are all invalid."
+        )
+        # TODO: refactor this so that AWS is not a "special" platform
         if config.ENV_DATA['platform'].lower() == constants.AWS_PLATFORM:
             if config.ENV_DATA['deployment_type'] == 'ipi':
-                node.delete_and_create_osd_node_aws_ipi(osd_node_name)
+                node.delete_and_create_osd_node_ipi(osd_node_name)
 
             elif config.ENV_DATA['deployment_type'] == 'upi':
                 node.delete_and_create_osd_node_aws_upi(osd_node_name)
             else:
-                pytest.fail(
-                    f"ocs-ci config 'deployment_type' value '{config.ENV_DATA['deployment_type']}' is not valid, "
-                    f"results of this test run are all invalid.")
-
+                log.error(msg_invalid)
+                pytest.fail(msg_invalid)
+        elif config.ENV_DATA['platform'].lower() in constants.CLOUD_PLATFORMS:
+            if config.ENV_DATA['deployment_type'] == 'ipi':
+                node.delete_and_create_osd_node_ipi(osd_node_name)
+            else:
+                log.error(msg_invalid)
+                pytest.fail(msg_invalid)
```

Comment:
do we need to report a failure if it's UPI? How about skip?

Reply:
Makes sense, but we need to make this check 1st step in the test.

Reply:
Thinking bit more: I don't think it makes sense to handle such case, as we don't plan to support UPI on other cloud platforms but AWS, so this should be never executed. And when it is (by accident?), reporting that this is invalid deployment seems to be accurate.

Reply:
Yes, I understand, this code should not be reached and if it does, a wrong configuration was used (like GCP UPI).
Good point

Reply:
One more comment, I think the conditions here should be changed, in order to support on prem IPI other than vSphere. RHV and BM for example (yes, BM will be IPI supported as well at some point).
For that, we can change the conditions to something like:

```python
if config.ENV_DATA['deployment_type'] == 'ipi':
    node.delete_and_create_osd_node_ipi(osd_node_name)
elif config.ENV_DATA['deployment_type'] == 'upi':
    if config.ENV_DATA['platform'].lower() in constants.CLOUD_PLATFORMS:
                log.error(msg_invalid)
                pytest.fail(msg_invalid)
    elif config.ENV_DATA['platform'].lower() == constants.VSPHERE_PLATFORM:
        pytest.skip("Skipping add node in Vmware platform due to "https://bugzilla.redhat.com/show_bug.cgi?id=1844521")
    else:
        pytest.skip("")
else:
    pytest.skip("Deployment type is not IPI or UPI")

   
```

Reply:
this could be done in a follow-up PR. Merging this one so this test will be executed over GCP and Azure.
Opened https://github.com/red-hat-storage/ocs-ci/issues/2927 for that

Reply:
Agreed, the overall structure should be changed as you drafted.